tests/storage/test_base.py

```diff
@@ -146,7 +146,7 @@ def test_select_one_missing(
     @defer.inlineCallbacks
     def test_select_list(self) -> Generator["defer.Deferred[object]", object, None]:
         self.mock_txn.rowcount = 3
-        self.mock_txn.__iter__ = Mock(return_value=iter([(1,), (2,), (3,)]))
+        self.mock_txn.fetchall.return_value = [(1,), (2,), (3,)]
```

Comment:
To check: I _think_ that this is needed because  we've changed the implementation to use `fetchall` (rather than a dictionary comprehension that loops over the psycopg cursor). Do I understand correctly?

Reply:
Yes, exactly! 👍